### PR TITLE
viennacl 1.6.2 (new formula)

### PR DESCRIPTION
**Pull request by dpalmer93**

### CODE REVIEW

**Comment by bfontaine** [comment]
The test doesn’t pass: http://bot.brew.sh/job/Homebrew%20Pull%20Requests/29600/version=mountain_lion/testReport/junit/brew-test-bot/mountain_lion/test_viennacl/

File: Library/Formula/viennacl.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
+class Viennacl < Formula
+  desc "ViennaCL is a free open-source linear algebra library for computations on many-core architectures (GPUs, MIC) and multi-core CPUs."
```
Comment (bfontaine) [inline Library/Formula/viennacl.rb]: Please reduce this, “viennacl: desc…” should fit in 80 characters.

File: Library/Formula/viennacl.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  homepage "http://viennacl.sourceforge.net"
+  url "https://downloads.sourceforge.net/project/viennacl/1.6.x/ViennaCL-1.6.2.tar.gz"
+  head "https://github.com/viennacl/viennacl-dev.git"
+  sha256 "e46e1960b0bd87619bf02a884611ca087ac3c7378b35d5be3d8185f4fe669805"
```
Comment (bfontaine) [inline Library/Formula/viennacl.rb]: Please move the `sha256` line above the `head` one.

**Comment by dpalmer93** [comment]
Fixed desc, sha256, and test.

**Comment by bfontaine** [comment]
The test is still failing:

```
libc++abi.dylib: terminating with uncaught exception of type viennacl::device_specific::generator_not_supported_exception: ViennaCL: Internal error: The generator cannot handle the statement provided: The supplied parameters for this template are invalid : err -4
Error: viennacl: failed
```

File: Library/Formula/viennacl.rb [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  head "https://github.com/viennacl/viennacl-dev.git"
+
+  depends_on "cmake" => :build
+  depends_on :macos => :snow_leopard
+
+  def install
+    system "cmake", ".", *std_cmake_args
+    system "make", "install"
+    prefix.install "#{buildpath}/examples/benchmarks/dense_blas-bench-cpu" => "test"
```
Comment (dpalmer93) [inline Library/Formula/viennacl.rb]: The old test seemed highly dependent on OpenCL configuration. Switched to a CPU-only test.

Comment (DomT4) [inline Library/Formula/viennacl.rb]: Let's place this in `libexec` instead of the `prefix`.

Comment (DomT4) [inline Library/Formula/viennacl.rb]: Or `pkgshare`. Either works.

**Comment by DomT4** [comment]
@BrewTestBot test this please

File: Library/Formula/viennacl.rb [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+  depends_on :macos => :snow_leopard
+
+  def install
+    system "cmake", ".", *std_cmake_args
+    system "make", "install"
+    libexec.install "#{buildpath}/examples/benchmarks/dense_blas-bench-cpu" => "test"
+  end
+
+  test do
+    system "#{opt_prefix}/libexec/test"
```
Comment (DomT4) [inline Library/Formula/viennacl.rb]: This can be `"#{opt_libexec}/test"`

**Comment by dunn** [comment]
Merged in 4ec4fa02720396df936cb9283136bbe10287b201, thanks so much!

**Pull request closed without merging**